### ISSUE TITLE
ollama: add native structured output support

### ISSUE THREAD

**Issue by hntrl** [langchain-ai/langchainjs#9353]
### Privileged issue

- [x] I am a LangChain maintainer, or was asked directly by a LangChain maintainer to create an issue here.

### Issue Content

Native json schema outputs has had support in Ollama for a while, but we're still relying on forced tool calling inside of `ChatOllama`. Different models will have different capabilities (e.g. only some models support tool calling + tool choice, so I imagine the same is applicable to structured outputs) so we'll need to introduce some kind of method selection based on a static model list to fully support this.

https://ollama.com/blog/structured-outputs

**Comment by chiichen** [comment]
>  (e.g. only some models support tool calling + tool choice, so I imagine the same is applicable to structured outputs) 

There's a little misunderstanding about structured outputs, unlike the ability of tool calling and tool choice(which is enabled by sft or rl) , actually it's only about the inference framework we used instead of the model itself. Here is a related issue #9333 , and this issue might be closed with it?

**Comment by ItzCrazyKns** [comment]
> ### Privileged issue
> * [x]  I am a LangChain maintainer, or was asked directly by a LangChain maintainer to create an issue here.
> 
> ### Issue Content
> Native json schema outputs has had support in Ollama for a while, but we're still relying on forced tool calling inside of `ChatOllama`. Different models will have different capabilities (e.g. only some models support tool calling + tool choice, so I imagine the same is applicable to structured outputs) so we'll need to introduce some kind of method selection based on a static model list to fully support this.
> 
> https://ollama.com/blog/structured-outputs

Structured outputs in Ollama work by applying a grammar mask to the model’s logits during generation so the model can only produce tokens that match the specified JSON schema and this makes the feature completely model agnostic because it operates after the logits are computed rather than relying on any built in tool calling or special model capabilities, so it would be ideal for LangChain to add native support for structured outputs directly instead of forcing tool usage since the grammar constrained generation already guarantees valid JSON for all models rather than issues that can be caused by tool calling & tool choice

**Comment by christian-bromann** [comment]
This was fixed and https://github.com/langchain-ai/langchainjs/pull/9580 and will be published with the upcoming release.
